docs/examples/auto-instrumentation/README.rst

```diff
@@ -1,209 +0,0 @@
-Auto-instrumentation
```

Comment:
I've removed the Auto Instrumentation example here because I'm not sure if we want it in the `opentelemetry-instrumentation` repo. Please let me know if we should move it there!